mongodbatlas/config.go

```diff
@@ -50,15 +47,30 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
 		return nil, diag.FromErr(err)
 	}
 
+	clients := &MongoDBClient{
+		Atlas:      atlasClient,
+		PublicKey:  c.PublicKey,
+		PrivateKey: c.PrivateKey,
+		BaseURL:    c.BaseURL,
+	}
+
+	return clients, nil
+}
+
+func (c *MongoDBClient) GetRealmClient(ctx context.Context) (*realm.Client, error) {
 	// Realm
+	if cast.ToString(c.PublicKey) == "" && cast.ToString(c.PrivateKey) == "" {
```

Comment:
Can you add the realm client as a private attribute and then cache it. Then you can just check if the client was already created instead of creating a new one every time this is called

`````
if client.realm != nil
  return client.real, nil

`````

Unless the token expires and we actually need to to this  every single time

Reply:
Would be great but the token usually expires after 5 minutes I think, so it's better to use it every single time